common/services/healthcheck.js

```diff
@@ -0,0 +1,52 @@
+const serviceCheckFactory = require('../data/healthcheck')
+
+const service = (name, url) => {
+  const check = serviceCheckFactory(name, url)
+  return () =>
+    check()
+      .then(result => ({ name, status: 'ok', message: result }))
+      .catch(err => ({ name, status: 'ERROR', message: err }))
+}
+
+module.exports = (auth, oasys, sentencePlanning, elite2) => {
+  const checks = [
+    service('auth', auth),
+    service('oasys', oasys),
+    service('sentencePlanning', sentencePlanning),
+    service('elite2', elite2),
+  ]
+
+  return callback =>
+    Promise.all(checks.map(fn => fn())).then(checkResults => {
+      const allOk = checkResults.every(item => item.status === 'ok')
+      const result = {
+        healthy: allOk,
+        checks: checkResults.reduce(gatherCheckInfo, {}),
+      }
+      callback(null, addAppInfo(result))
+    })
+}
+
+function gatherCheckInfo(total, currentValue) {
+  return { ...total, [currentValue.name]: currentValue.message }
+}
+
+function addAppInfo(result) {
+  const buildInformation = getBuild()
+  const buildInfo = {
+    uptime: process.uptime(),
+    build: buildInformation,
+    version: buildInformation && buildInformation.buildNumber,
+  }
+
+  return { ...result, ...buildInfo }
+}
+
+function getBuild() {
```

Comment:
meh.. is this never not present?  if it is could potentially put in a default file which is overwritten by the build?

Reply:
I agree about putting in a local default. I appreciate its an unlikely absentee, I'm inclined to just leave it though; added robustness against the improbable.